azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -178,20 +217,39 @@ def context(self) -> dict:
         return context
 
     def _init_impl(self):
-        try:
-            self.msi_auth_context = ManagedIdentityCredential(**self._msi_args)
-        except Exception as e:
-            raise KustoClientError("Failed to initialize MSI ManagedIdentityCredential with [" + str(self._msi_params) + "]\n" + str(e))
+        pass
 
     def _get_token_impl(self) -> dict:
-        return None
+        try:

```

Comment:
Catch by exception type or add a variable to decide which error message to throw.

Reply:
Sorry, I mean, instead of the try inside try..

Reply:
got it, should  be good now